src/main/java/com/example/glovo/repository/OrderRepository.java

```diff
@@ -0,0 +1,64 @@
+package com.example.glovo.repository;
+
+import com.example.glovo.dto.OrderDto;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class OrderRepository {
+
+    private List<OrderDto> orders = List.of(
+            OrderDto.builder().products(List.of("Bananas", "Bottle of water")).customerId(12).build(),
+            OrderDto.builder().products(List.of("Nuts", "Milk", "Bread")).customerId(25).build());
+
+    public OrderDto findById(int id) {
+        return orders.get(id);
+    }
+
+    public boolean save(OrderDto orderDto) {
+        ArrayList<OrderDto> newOrders = new ArrayList<>(orders);
```

Comment:
what is the reason to create new collection each time?

Reply:
It's better to use interface declaration like:
`List list = new ArrayList();` 
[](https://stackoverflow.com/questions/4991516/when-should-i-use-an-interface-vs-an-implementation-when-declaring-a-new-object)